snp_motion_planning/src/plugins/tasks/tcp_speed_limiter_task.cpp

```diff
@@ -0,0 +1,158 @@
+  {
+  }
+
+  explicit TCPSpeedLimiterTask(std::string name, std::string input_key, std::string output_key,
+                               bool is_conditional = true)
+    : tesseract_planning::TaskComposerTask(std::move(name), is_conditional)
+  {
+    input_keys_.push_back(std::move(input_key));
+    output_keys_.push_back(std::move(output_key));
+  }
+
+  explicit TCPSpeedLimiterTask(std::string name, const YAML::Node& config,
+                               const tesseract_planning::TaskComposerPluginFactory& /*plugin_factory*/)
+    : tesseract_planning::TaskComposerTask(std::move(name), config)
+  {
+    if (input_keys_.empty())
+      throw std::runtime_error("TCPSpeedLimiterTask, config missing 'inputs' entry");
+
+    if (input_keys_.size() > 1)
+      throw std::runtime_error("TCPSpeedLimiterTask, config 'inputs' entry currently only "
+                               "supports "
+                               "one input key");
+
+    if (output_keys_.empty())
+      throw std::runtime_error("TCPSpeedLimiterTask, config missing 'outputs' entry");
+
+    if (output_keys_.size() > 1)
+      throw std::runtime_error("TCPSpeedLimiterTask, config 'outputs' entry currently only "
+                               "supports "
+                               "one output key");
+  }
+
+  ~TCPSpeedLimiterTask() override = default;
+  TCPSpeedLimiterTask(const TCPSpeedLimiterTask&) = delete;
+  TCPSpeedLimiterTask& operator=(const TCPSpeedLimiterTask&) = delete;
+  TCPSpeedLimiterTask(TCPSpeedLimiter&&) = delete;
+  TCPSpeedLimiterTask& operator=(TCPSpeedLimiterTask&&) = delete;
+
+  bool operator==(const TCPSpeedLimiterTask& rhs) const
+  {
+    bool equal = true;
+    equal &= tesseract_planning::TaskComposerTask::operator==(rhs);
+    return equal;
+  }
+  bool operator!=(const TCPSpeedLimiterTask& rhs) const
+  {
+    return !operator==(rhs);
+  }
+
+protected:
+  friend struct tesseract_common::Serialization;
+  friend class boost::serialization::access;
+  tesseract_planning::TaskComposerNodeInfo::UPtr runImpl(tesseract_planning::TaskComposerContext& context,
+                                                         OptionalTaskComposerExecutor /*executor*/) const override final
+  {
+    auto info = std::make_unique<tesseract_planning::TaskComposerNodeInfo>(*this);
+    info->return_value = 0;
+
+    // Get the problem
+    auto& problem = dynamic_cast<tesseract_planning::PlanningTaskComposerProblem&>(*context.problem);
+
+    // --------------------
+    // Check that inputs are valid
+    // --------------------
+    auto input_data_poly = context.data_storage->getData(input_keys_[0]);
+    if (input_data_poly.isNull() ||
+        input_data_poly.getType() != std::type_index(typeid(tesseract_planning::CompositeInstruction)))
+    {
+      info->message = "Input results to TCP speed limiter task must be a composite instruction";
+      CONSOLE_BRIDGE_logError("%s", info->message.c_str());
```

Comment:
Remove; let some other part of the program log this so we don't clutter the terminal